app/src/test/java/com/alphawallet/app/ENSTest.java

```diff
@@ -103,16 +105,46 @@ public void testReverseResolve() throws Exception {
 
     @Test
     public void testNameHash() {
-        assertEquals(
-                nameHash(""),
-                ("0x0000000000000000000000000000000000000000000000000000000000000000"));
-
-        assertEquals(
-                nameHash("eth"),
-                ("0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae"));
+        try
+        {
+            assertEquals(
+                    nameHash(""),
+                    ("0x0000000000000000000000000000000000000000000000000000000000000000"));
+
+            assertEquals(
+                    nameHash("eth"),
+                    ("0x93cdeb708b7545dc668eb9280176169d1c33cfd8ed6f04690a0bcc88a93fc4ae"));
+
+            assertEquals(
+                    nameHash("foo.eth"),
+                    ("0xde9b09fd7c5f901e23a3f19fecc54828e9c848539801e86591bd9801b019f84f"));
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+            fail();
+        }
+    }
 
-        assertEquals(
-                nameHash("foo.eth"),
-                ("0xde9b09fd7c5f901e23a3f19fecc54828e9c848539801e86591bd9801b019f84f"));
+    @Test
+    public void testNPE()
+    {
+        try
+        {
+            String test = nameHash(null);
+            System.out.println("Should have triggered exception");
+            fail();
+        }
+        catch (EnsResolutionException e)
+        {
+            System.out.println("Correctly got the exception");
+            e.printStackTrace();
+        }
+        catch (Exception e)
+        {
+            //should be ENSResolutionException
+            System.out.println("Should be EnsResoltionException");
+            fail();
+        }
```

Comment:
In JUnit 4, can test Exception this way:
```
@Rule
public ExpectedException exceptionRule = ExpectedException.none();

@Test
public void whenExceptionThrown_thenRuleIsApplied() {
    exceptionRule.expect(NumberFormatException.class);
    exceptionRule.expectMessage("For input string");
    Integer.parseInt("1a");
}
```

Reply:
The source said the ExpectedException syntax is deprecated, use 'Assert.assertThrows' syntax instead.